src/lib/API/api.js

```diff
@@ -149,6 +149,15 @@ export class APIService {
     1000,
   )
 
+  /**
+   * `/user/start` post api call
+   * @param {UserRecord} user
+   */
+  addSignupContact = throttle(
```

Comment:
i'm not sure it's a good idea to throttle/debounce here. 
the api class should provide clear interface with one call - one action. e.g. if we called something - it **should be** executed immediately. 
so for the better design the throttling have to be moved out this class

i propose to do the following:
- add src/lib/hooks/useThrottle.js
```javascript
import { useState, useEffect, useRef } from 'react';

export const useThrottle = (value, limit) => {
  const [throttledValue, setThrottledValue] = useState(value);
  const lastRan = useRef(Date.now());

  useEffect(
    () => {
      const handler = setTimeout(function() {
        if (Date.now() - lastRan.current >= limit) {
          setThrottledValue(value);
          lastRan.current = Date.now();
        }
      }, limit - (Date.now() - lastRan.current));

      return () => {
        clearTimeout(handler);
      };
    },
    [value, limit]
  );

  return throttledValue;
};
```
- throttle state.email changes
```
const throttledEmail = useThrottle(state.email, 1000)
```
- use on change effect on throttledEmail
```javascript
// listening to the email changes in the state
  useEffect(() => {
    // perform this again for torus and on email change. torus has also mobile verification that doesnt set email
    if (!throttledEmail) {
      return
    }

     // once email appears in the state - identifying and setting 'identified' flag
     identifyOnUserSignup(email)
     API.addSignupContact(state)
       .then(r => log.info('addSignupContact success', { state }))
       .catch(e => log.error('addSignupContact failed', e.message, e))
   }, [throttledEmail])
```

Reply:
there's no reason for the api method to be called multiple times quickly so its perfectly fine to put that guard in the api

Reply:
> there's no reason for the api method to be called multiple times quickly so its perfectly fine to put that guard in the api

i don't agree with that because it's violates with SOLID, single responsibility principle particularly
api class should interact with server, useThrottle hook should add call rate guard. one component - one task

but ok, let it be like that for now. approving PR